File: frontend/app/models/organisation.ts
import { Address } from 'app/models/address';
import { Model } from 'app/models/model';

export class Organisation extends Model {

	public name: string = '';
	public description: string = '';
	public website: string = '';
	public mail: string = '';
	public phone: string = '';
	public image: any = null;
	public address_id: string = '';
	public address: Address = new Address({});
	public _translations: any = {};


	constructor(json: any = {} as Organisation) {
		super(json.id);
		this.name = json.name && json.name || '';
		this.description = json.description && json.description || '';
		this.website = json.website && json.website || '';
		this.mail = json.mail && json.mail || '';
		this.phone = json.phone && json.phone || '';
		this.image = json.image && json.image || null;
		this.address_id = json.address_id && json.address_id || {};
		this._translations = json._translations && Object.assign(json._translations) || {};
		if (json.address) { this.address = new Address(json.address); }

	}

}

File: server/wupportal/src/main/java/de/codeschluss/wupportal/model/Organisation.java
package de.codeschluss.wupportal.model;

import java.io.Serializable;
import java.util.List;

import javax.persistence.*;

/**
 * The persistent class for the organisations database table.
 * 
 */
@Entity
@Table(name = "organisations")
public class Organisation extends BaseEntity implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private String id;

	@Lob
	private String description;

	@Lob
	private byte[] image;

	private String mail;

	private String name;

	private String phone;

	private String website;

	// bi-directional many-to-one association to Address
	@ManyToOne
	private Address address;

	@OneToMany(mappedBy = "organisation")
	private List<Provider> providers;

	public Organisation() {
	}

	public String getId() {
		return this.id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getDescription() {
		return this.description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public byte[] getImage() {
		return this.image;
	}

	public void setImage(byte[] image) {
		this.image = image;
	}

	public String getMail() {
		return this.mail;
	}

	public void setMail(String mail) {
		this.mail = mail;
	}

	public String getName() {
		return this.name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getPhone() {
		return this.phone;
	}

	public void setPhone(String phone) {
		this.phone = phone;
	}

	public String getWebsite() {
		return this.website;
	}

	public void setWebsite(String website) {
		this.website = website;
	}

	public Address getAddress() {
		return this.address;
	}

	public void setAddress(Address address) {
		this.address = address;
	}

	public List<Provider> getProviders() {
		return this.providers;
	}

	public void setProviders(List<Provider> providers) {
		this.providers = providers;
	}

}